### PR TITLE
Don't crash on missing fields in pull requests

### PR DESCRIPTION
**Pull request by emptywee**
Either GitHub changed something on their API end, or something else happened, but sometimes there's no comment_data or commit_data fields available in pull request response from GitHub API. Moreover, such requests come as dictionary wrapped into a list:

(redacted response)

```
[
    {
        "_links": {
            "comments": {
                "href": "https://api.github.com/repos/org_name/some_repo/issues/1/comments"
            },
            "commits": {
                "href": "https://api.github.com/repos/org_name/some_repo/pulls/1/commits"
            },
            "html": {
                "href": "https://github.com/org_name/some_repo/pull/1"
            },
            "issue": {
                "href": "https://api.github.com/repos/org_name/some_repo/issues/1"
            },
            "review_comment": {
                "href": "https://api.github.com/repos/org_name/some_repo/pulls/comments{/number}"
            },
            "review_comments": {
                "href": "https://api.github.com/repos/org_name/some_repo/pulls/1/comments"
            },
            "self": {
                "href": "https://api.github.com/repos/org_name/some_repo/pulls/1"
            },
            "statuses": {
                "href": "https://api.github.com/repos/org_name/some_repo/statuses/0b2ea3eeb4c03e48c9c7b1************"
            }
        },
        "additions": 1845,
        "assignee": null,
        "assignees": [],
        "author_association": "COLLABORATOR",
        "base": {
            "label": "org_name:master",
            "ref": "master",
            "repo": {
                "archive_url": "https://api.github.com/repos/org_name/some_repo/{archive_format}{/ref}",
                "archived": false,
                "assignees_url": "https://api.github.com/repos/org_name/some_repo/assignees{/user}",
                "blobs_url": "https://api.github.com/repos/org_name/some_repo/git/blobs{/sha}",
                "branches_url": "https://api.github.com/repos/org_name/some_repo/branches{/branch}",
                "clone_url": "https://github.com/org_name/some_repo.git",
                "collaborators_url": "https://api.github.com/repos/org_name/some_repo/collaborators{/collaborator}",
                "comments_url": "https://api.github.com/repos/org_name/some_repo/comments{/number}",
                "commits_url": "https://api.github.com/repos/org_name/some_repo/commits{/sha}",
                "compare_url": "https://api.github.com/repos/org_name/some_repo/compare/{base}...{head}",
                "contents_url": "https://api.github.com/repos/org_name/some_repo/contents/{+path}",
                "contributors_url": "https://api.github.com/repos/org_name/some_repo/contributors",
                "created_at": "2017-01-30T22:17:29Z",
                "default_branch": "master",
                "deployments_url": "https://api.github.com/repos/org_name/some_repo/deployments",
                "description": "some_repo",
                "downloads_url": "https://api.github.com/repos/org_name/some_repo/downloads",
                "events_url": "https://api.github.com/repos/org_name/some_repo/events",
                "fork": false,
                "forks": 0,
                "forks_count": 0,
                "forks_url": "https://api.github.com/repos/org_name/some_repo/forks",
                "full_name": "org_name/some_repo",
                "git_commits_url": "https://api.github.com/repos/org_name/some_repo/git/commits{/sha}",
                "git_refs_url": "https://api.github.com/repos/org_name/some_repo/git/refs{/sha}",
                "git_tags_url": "https://api.github.com/repos/org_name/some_repo/git/tags{/sha}",
                "git_url": "git://github.com/org_name/some_repo.git",
                "has_downloads": true,
                "has_issues": true,
                "has_pages": false,
                "has_projects": true,
                "has_wiki": true,
                "homepage": null,
                "hooks_url": "https://api.github.com/repos/org_name/some_repo/hooks",
                "html_url": "https://github.com/org_name/some_repo",
                "id": 80469388,
                "issue_comment_url": "https://api.github.com/repos/org_name/some_repo/issues/comments{/number}",
                "issue_events_url": "https://api.github.com/repos/org_name/some_repo/issues/events{/number}",
                "issues_url": "https://api.github.com/repos/org_name/some_repo/issues{/number}",
                "keys_url": "https://api.github.com/repos/org_name/some_repo/keys{/key_id}",
                "labels_url": "https://api.github.com/repos/org_name/some_repo/labels{/name}",
                "language": "Java",
                "languages_url": "https://api.github.com/repos/org_name/some_repo/languages",
                "license": null,
                "merges_url": "https://api.github.com/repos/org_name/some_repo/merges",
                "milestones_url": "https://api.github.com/repos/org_name/some_repo/milestones{/number}",
                "mirror_url": null,
                "name": "some_repo",
                "node_id": "nodeid_base64",
                "notifications_url": "https://api.github.com/repos/org_name/some_repo/notifications{?since,all,participating}",
                "open_issues": 1,
                "open_issues_count": 1,
                "owner": {
                    "avatar_url": "https://avatars1.githubusercontent.com/u/user_id?v=4",
                    "events_url": "https://api.github.com/users/org_name/events{/privacy}",
                    "followers_url": "https://api.github.com/users/org_name/followers",
                    "following_url": "https://api.github.com/users/org_name/following{/other_user}",
                    "gists_url": "https://api.github.com/users/org_name/gists{/gist_id}",
                    "gravatar_id": "",
                    "html_url": "https://github.com/org_name",
                    "id": user_id,
                    "login": "org_name",
                    "node_id": "nodeid_base64",
                    "organizations_url": "https://api.github.com/users/org_name/orgs",
                    "received_events_url": "https://api.github.com/users/org_name/received_events",
                    "repos_url": "https://api.github.com/users/org_name/repos",
                    "site_admin": false,
                    "starred_url": "https://api.github.com/users/org_name/starred{/owner}{/repo}",
                    "subscriptions_url": "https://api.github.com/users/org_name/subscriptions",
                    "type": "Organization",
                    "url": "https://api.github.com/users/org_name"
                },
                "private": true,
                "pulls_url": "https://api.github.com/repos/org_name/some_repo/pulls{/number}",
                "pushed_at": "2018-08-27T21:59:22Z",
                "releases_url": "https://api.github.com/repos/org_name/some_repo/releases{/id}",
                "size": 1366,
                "ssh_url": "git@github.com:org_name/some_repo.git",
                "stargazers_count": 0,
                "stargazers_url": "https://api.github.com/repos/org_name/some_repo/stargazers",
                "statuses_url": "https://api.github.com/repos/org_name/some_repo/statuses/{sha}",
                "subscribers_url": "https://api.github.com/repos/org_name/some_repo/subscribers",
                "subscription_url": "https://api.github.com/repos/org_name/some_repo/subscription",
                "svn_url": "https://github.com/org_name/some_repo",
                "tags_url": "https://api.github.com/repos/org_name/some_repo/tags",
                "teams_url": "https://api.github.com/repos/org_name/some_repo/teams",
                "trees_url": "https://api.github.com/repos/org_name/some_repo/git/trees{/sha}",
                "updated_at": "2018-08-10T15:54:44Z",
                "url": "https://api.github.com/repos/org_name/some_repo",
                "watchers": 0,
                "watchers_count": 0
            },
            "sha": "0e166fdd81592fec3e88de2c2dd49b39993170cf",
            "user": {
                "avatar_url": "https://avatars1.githubusercontent.com/u/id?v=4",
                "events_url": "https://api.github.com/users/org_name/events{/privacy}",
                "followers_url": "https://api.github.com/users/org_name/followers",
                "following_url": "https://api.github.com/users/org_name/following{/other_user}",
                "gists_url": "https://api.github.com/users/org_name/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/org_name",
                "id": id,
                "login": "org_name",
                "node_id": "node_id",
                "organizations_url": "https://api.github.com/users/org_name/orgs",
                "received_events_url": "https://api.github.com/users/org_name/received_events",
                "repos_url": "https://api.github.com/users/org_name/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/org_name/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/org_name/subscriptions",
                "type": "Organization",
                "url": "https://api.github.com/users/org_name"
            }
        },
        "body": "",
        "changed_files": 32,
        "closed_at": null,
        "comments": 0,
        "comments_url": "https://api.github.com/repos/org_name/some_repo/issues/1/comments",
        "commits": 24,
        "commits_url": "https://api.github.com/repos/org_name/some_repo/pulls/1/commits",
        "created_at": "2018-08-21T17:06:39Z",
        "deletions": 104,
        "diff_url": "https://github.com/org_name/some_repo/pull/1.diff",
        "head": {
            "label": "org_name:EXAMPLEORG",
            "ref": "EXAMPLEORG",
            "repo": {
                "archive_url": "https://api.github.com/repos/org_name/some_repo/{archive_format}{/ref}",
                "archived": false,
                "assignees_url": "https://api.github.com/repos/org_name/some_repo/assignees{/user}",
                "blobs_url": "https://api.github.com/repos/org_name/some_repo/git/blobs{/sha}",
                "branches_url": "https://api.github.com/repos/org_name/some_repo/branches{/branch}",
                "clone_url": "https://github.com/org_name/some_repo.git",
                "collaborators_url": "https://api.github.com/repos/org_name/some_repo/collaborators{/collaborator}",
                "comments_url": "https://api.github.com/repos/org_name/some_repo/comments{/number}",
                "commits_url": "https://api.github.com/repos/org_name/some_repo/commits{/sha}",
                "compare_url": "https://api.github.com/repos/org_name/some_repo/compare/{base}...{head}",
                "contents_url": "https://api.github.com/repos/org_name/some_repo/contents/{+path}",
                "contributors_url": "https://api.github.com/repos/org_name/some_repo/contributors",
                "created_at": "2017-01-30T22:17:29Z",
                "default_branch": "master",
                "deployments_url": "https://api.github.com/repos/org_name/some_repo/deployments",
                "description": "some_repo",
                "downloads_url": "https://api.github.com/repos/org_name/some_repo/downloads",
                "events_url": "https://api.github.com/repos/org_name/some_repo/events",
                "fork": false,
                "forks": 0,
                "forks_count": 0,
                "forks_url": "https://api.github.com/repos/org_name/some_repo/forks",
                "full_name": "org_name/some_repo",
                "git_commits_url": "https://api.github.com/repos/org_name/some_repo/git/commits{/sha}",
                "git_refs_url": "https://api.github.com/repos/org_name/some_repo/git/refs{/sha}",
                "git_tags_url": "https://api.github.com/repos/org_name/some_repo/git/tags{/sha}",
                "git_url": "git://github.com/org_name/some_repo.git",
                "has_downloads": true,
                "has_issues": true,
                "has_pages": false,
                "has_projects": true,
                "has_wiki": true,
                "homepage": null,
                "hooks_url": "https://api.github.com/repos/org_name/some_repo/hooks",
                "html_url": "https://github.com/org_name/some_repo",
                "id": id,
                "issue_comment_url": "https://api.github.com/repos/org_name/some_repo/issues/comments{/number}",
                "issue_events_url": "https://api.github.com/repos/org_name/some_repo/issues/events{/number}",
                "issues_url": "https://api.github.com/repos/org_name/some_repo/issues{/number}",
                "keys_url": "https://api.github.com/repos/org_name/some_repo/keys{/key_id}",
                "labels_url": "https://api.github.com/repos/org_name/some_repo/labels{/name}",
                "language": "Java",
                "languages_url": "https://api.github.com/repos/org_name/some_repo/languages",
                "license": null,
                "merges_url": "https://api.github.com/repos/org_name/some_repo/merges",
                "milestones_url": "https://api.github.com/repos/org_name/some_repo/milestones{/number}",
                "mirror_url": null,
                "name": "some_repo",
                "node_id": "node_id",
                "notifications_url": "https://api.github.com/repos/org_name/some_repo/notifications{?since,all,participating}",
                "open_issues": 1,
                "open_issues_count": 1,
                "owner": {
                    "avatar_url": "https://avatars1.githubusercontent.com/u/user_id?v=4",
                    "events_url": "https://api.github.com/users/org_name/events{/privacy}",
                    "followers_url": "https://api.github.com/users/org_name/followers",
                    "following_url": "https://api.github.com/users/org_name/following{/other_user}",
                    "gists_url": "https://api.github.com/users/org_name/gists{/gist_id}",
                    "gravatar_id": "",
                    "html_url": "https://github.com/org_name",
                    "id": id,
                    "login": "org_name",
                    "node_id": "node_id",
                    "organizations_url": "https://api.github.com/users/org_name/orgs",
                    "received_events_url": "https://api.github.com/users/org_name/received_events",
                    "repos_url": "https://api.github.com/users/org_name/repos",
                    "site_admin": false,
                    "starred_url": "https://api.github.com/users/org_name/starred{/owner}{/repo}",
                    "subscriptions_url": "https://api.github.com/users/org_name/subscriptions",
                    "type": "Organization",
                    "url": "https://api.github.com/users/org_name"
                },
                "private": true,
                "pulls_url": "https://api.github.com/repos/org_name/some_repo/pulls{/number}",
                "pushed_at": "2018-08-27T21:59:22Z",
                "releases_url": "https://api.github.com/repos/org_name/some_repo/releases{/id}",
                "size": 1366,
                "ssh_url": "git@github.com:org_name/some_repo.git",
                "stargazers_count": 0,
                "stargazers_url": "https://api.github.com/repos/org_name/some_repo/stargazers",
                "statuses_url": "https://api.github.com/repos/org_name/some_repo/statuses/{sha}",
                "subscribers_url": "https://api.github.com/repos/org_name/some_repo/subscribers",
                "subscription_url": "https://api.github.com/repos/org_name/some_repo/subscription",
                "svn_url": "https://github.com/org_name/some_repo",
                "tags_url": "https://api.github.com/repos/org_name/some_repo/tags",
                "teams_url": "https://api.github.com/repos/org_name/some_repo/teams",
                "trees_url": "https://api.github.com/repos/org_name/some_repo/git/trees{/sha}",
                "updated_at": "2018-08-10T15:54:44Z",
                "url": "https://api.github.com/repos/org_name/some_repo",
                "watchers": 0,
                "watchers_count": 0
            },
            "sha": "0b2ea3eeb4c03e48c9c7b1bc344603e44ce07570",
            "user": {
                "avatar_url": "https://avatars1.githubusercontent.com/u/user_id?v=4",
                "events_url": "https://api.github.com/users/org_name/events{/privacy}",
                "followers_url": "https://api.github.com/users/org_name/followers",
                "following_url": "https://api.github.com/users/org_name/following{/other_user}",
                "gists_url": "https://api.github.com/users/org_name/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/org_name",
                "id": user_id,
                "login": "org_name",
                "node_id": "node_id",
                "organizations_url": "https://api.github.com/users/org_name/orgs",
                "received_events_url": "https://api.github.com/users/org_name/received_events",
                "repos_url": "https://api.github.com/users/org_name/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/org_name/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/org_name/subscriptions",
                "type": "Organization",
                "url": "https://api.github.com/users/org_name"
            }
        },
        "html_url": "https://github.com/org_name/some_repo/pull/1",
        "id": 209894935,
        "issue_url": "https://api.github.com/repos/org_name/some_repo/issues/1",
        "labels": [],
        "locked": false,
        "maintainer_can_modify": false,
        "merge_commit_sha": null,
        "mergeable": false,
        "mergeable_state": "dirty",
        "merged": false,
        "merged_at": null,
        "merged_by": null,
        "milestone": null,
        "node_id": "node_id",
        "number": 1,
        "patch_url": "https://github.com/org_name/some_repo/pull/1.patch",
        "rebaseable": false,
        "requested_reviewers": [],
        "requested_teams": [],
        "review_comment_url": "https://api.github.com/repos/org_name/some_repo/pulls/comments{/number}",
        "review_comments": 0,
        "review_comments_url": "https://api.github.com/repos/org_name/some_repo/pulls/1/comments",
        "state": "open",
        "statuses_url": "https://api.github.com/repos/org_name/some_repo/statuses/0b2ea3eeb4c03e48c9c7b***********",
        "title": "EXAMPLEORG",
        "updated_at": "2018-08-27T21:59:23Z",
        "url": "https://api.github.com/repos/org_name/some_repo/pulls/1",
        "user": {
            "avatar_url": "https://avatars2.githubusercontent.com/u/uid?v=4",
            "events_url": "https://api.github.com/users/username1/events{/privacy}",
            "followers_url": "https://api.github.com/users/username1/followers",
            "following_url": "https://api.github.com/users/username1/following{/other_user}",
            "gists_url": "https://api.github.com/users/username1/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/username1",
            "id": uid,
            "login": "username1",
            "node_id": "node_id",
            "organizations_url": "https://api.github.com/users/username1/orgs",
            "received_events_url": "https://api.github.com/users/username1/received_events",
            "repos_url": "https://api.github.com/users/username1/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/username1/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/username1/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/username1"
        }
    }
]
```

This PR is to at least skip such pull requests and not crash.

### CODE REVIEW

**Comment by emptywee** [comment]
Previously, the script would crash with:

```
Traceback (most recent call last):
  File "/usr/local/bin/github-backup", line 991, in <module>
    main()
  File "/usr/local/bin/github-backup", line 986, in main
    backup_repositories(args, output_directory, repositories)
  File "/usr/local/bin/github-backup", line 648, in backup_repositories
    backup_pulls(args, repo_cwd, repository, repos_template)
  File "/usr/local/bin/github-backup", line 770, in backup_pulls
    pulls[number]['comment_data'] = retrieve_data(args, template)
TypeError: list indices must be integers, not str

```

and same for `commit_data`

File: CHANGES.rst [hunk 1]
```diff
@@ -1,6 +1,14 @@
 Changelog
 =========
 
+0.20.1 (2018-08-28)
```
Comment (josegonzalez) [inline CHANGES.rst]: Please revert all the release related code, that is handled when I make a release.

Comment (emptywee) [reply CHANGES.rst]: Ooops, my bad. Will do.

**Comment by emptywee** [comment]
@josegonzalez release related changes have been reverted. Sorry once again.

**Comment by josegonzalez** [comment]
Mind squashing your commits together? I'll merge/release once that is done.

**Pull request closed without merging**